hera_pspec/grouping.py

```diff
@@ -348,66 +370,100 @@ def average_spectra(uvp_in, blpair_groups=None, time_avg=False,
+    stats_array = odict([[stat, odict()] for stat in stat_l])
+
+    # will average covariance array by default
+    store_cov = hasattr(uvp,"cov_array")
+    if store_cov:
+        cov_array = odict()
+
 
     # Iterate over spectral windows
     for spw in range(uvp.Nspws):
         spw_data, spw_wgts, spw_ints, spw_nsmp = [], [], [], []
-
+        spw_stats = odict([[stat, []] for stat in stat_l])
+        if store_cov:
+            spw_cov = []
         # Iterate over polarizations
         for i, p in enumerate(uvp.pol_array):
             pol_data, pol_wgts, pol_ints, pol_nsmp = [], [], [], []
+            pol_stats = odict([[stat, []] for stat in stat_l])
+            if store_cov:
+                pol_cov = []
 
             # Iterate over baseline-pair groups
             for j, blpg in enumerate(blpair_groups):
                 bpg_data, bpg_wgts, bpg_ints, bpg_nsmp = [], [], [], []
+                bpg_stats = odict([[stat, []] for stat in stat_l])
+                if store_cov:
+                    bpg_cov = []
                 w_list = []
-                
-                # Sum over all weights within this baseline group to get 
-                # normalization (if weights specified). The normalization is 
+
+                # Sum over all weights within this baseline group to get
+                # normalization (if weights specified). The normalization is
                 # calculated so that Sum (blpair wgts) = no. baselines.
                 if blpair_weights is not None:
                     blpg_wgts = np.array(blpair_weights[j])
                     norm = np.sum(blpg_wgts) if normalize_weights else 1.
-                    
+
                     if norm <= 0.:
                         raise ValueError("Sum of baseline-pair weights in "
                                          "group %d is <= 0." % j)
                     blpg_wgts *= float(blpg_wgts.size) / norm # Apply normalization
                 else:
                     blpg_wgts = np.ones(len(blpg))
-                
+
                 # Iterate within a baseline-pair group and get integration-
                 # weighted data
                 for k, blp in enumerate(blpg):
-                    
+
                     # Get no. samples and construct integration weight
                     nsmp = uvp.get_nsamples(spw, blp, p)[:, None]
                     data = uvp.get_data(spw, blp, p)
                     wgts = uvp.get_wgts(spw, blp, p)
                     ints = uvp.get_integrations(spw, blp, p)[:, None]
                     w = (ints * np.sqrt(nsmp))
-                    
+                    if store_cov:
+                        cov = uvp.get_cov(spw,blp,p)
+                    # Get error bar weights and set invalid ones to zero.
+                    errws = {}
+                    for stat in stat_l:
+                        errs = uvp.get_stats(stat, (spw, blp, p)).copy()
+                        no_data = np.isnan(errs)
```

Comment:
I think this should be
```
no_data = np.isnan(errs) + (errs < 0)
```
just to guard against weird cases when `errs` is negative, which I don't think should ever be the case.